packages/runtime/src/adapter/index.ts

```diff
@@ -0,0 +1,133 @@
+import type * as RDF from '@rdfjs/types'
+import { CONSTRUCT, Construct, Describe } from '@tpluscode/sparql-builder'
+import { DESCRIBE } from '@tpluscode/sparql-builder'
+import { VALUES } from '@tpluscode/sparql-builder/expressions'
+import * as D from 'docmaps-sdk'
+import * as TE from 'fp-ts/lib/TaskEither'
+import n3 from 'n3'
+import { prefixes } from '@zazuko/rdf-vocabularies'
+import namespace from '@rdfjs/namespace'
+import { collect } from 'streaming-iterables'
+import { pipe } from 'fp-ts/lib/pipeable'
+import factory from '@rdfjs/data-model'
+
+/* Inteface for an in-memory or over-the-web mechanism that accepts
+ * SPARQL queries and returns triples.
+ * @since 0.1.0
+ */
+export interface SparqlProcessor {
+  // truthy(query: string): Promise<boolean>
+  triples(query: Construct | Describe): Promise<AsyncIterable<RDF.Quad>>
+  // bindings(query: string): Promise<AsyncIterable<{ [key: string]: RDF.Term }>> // TODO: confirm how to handle generics better here
+}
+
+function FindDocmapQuery(iri: string): Construct | Describe {
+  const subj = factory.namedNode(iri)
+
+  const dcterms = namespace('http://purl.org/dc/terms/')
+  const pwo = namespace('http://purl.org/spar/pwo/')
+  const fabio = namespace('http://purl.org/spar/fabio/')
+  const pso = namespace('http://purl.org/spar/pso/')
+  const pro = namespace('http://purl.org/spar/pro/')
+  const tx = namespace('http://www.ontologydesignpatterns.org/cp/owl/taskexecution.owl#')
+
+  const values = [{ map: subj }]
+
+  const q = CONSTRUCT`
+    ?s ?p ?o
+  `.WHERE`
+    SELECT DISTINCT ?s ?p ?o
+    WHERE {
+      ?s ?p ?o .
+    }
```

Comment:
This is the beginning of a new mechanism. Using a nested SELECT DISTINCT improves performance by making the query engine deduplicate for us. However, it remains to be seen how to use SELECT to find an arbitrary number of variables. This CONSTRUCTion also passes tests, but only because it currently returns the whole dataset and there is only 1 docmap in the dataset. It would fail if the resulting triples represented multiple docmaps which would be the case in any realistic scenario.